### PR TITLE
Removed long deprecated code (older than 10 months)

### PR DESCRIPTION
**Pull request by adityapatruni**
### Description
Issue #1933 removed long deprecated functions that are obsolete and mentioned as not to be used.

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_core/robot_state/include/moveit/robot_state/robot_state.h [hunk 1]
```diff
@@ -1618,7 +1547,7 @@ class RobotState
    * the body positions will get corrupted.  You need to make a fresh
    * copy of the AttachedBody object for each RobotState you attach it
    * to.*/
```
Comment (AndyZe) [inline moveit_core/robot_state/include/moveit/robot_state/robot_state.h:1549]: Don't forget to delete the text above this, too

**Comment by AndyZe** [comment]
If you click on "Show all checks" above, you'll see that some of the CI tests failed. For example, click "Details" about Formatting and you'll see that a few empty lines need to be deleted.

**Review by henningkayser — COMMENTED** [review]
Thank you! Several of these functions are still defined in the source files which is why CI fails for the most part. The definitions need to be removed as well.

**Comment by adityapatruni** [comment]
Sure yes. I will do that when I get the chance. Just wanted to confirm if these function headers that are being removed are good to be deprecated fully. I have check the source code for references and confirmed that they are bot being used in other parts of the code.


On Tuesday, February 7, 2023, 11:13, Henning Kayser ***@***.***> wrote:




@henningkayser commented on this pull request.

Thank you! Several of these functions are still defined in the source files which is why CI fails for the most part. The definitions need to be removed as well.

**Comment by AndyZe** [comment]
> Just wanted to confirm if these function headers that are being removed are good to be deprecated fully.

Yep, go ahead and delete this stuff fully.

File: moveit_core/planning_scene/include/moveit/planning_scene/planning_scene.h [hunk 2]
```diff
@@ -920,20 +920,6 @@ class MOVEIT_PLANNING_SCENE_EXPORT PlanningScene : public std::enable_shared_fro
...
-  isEmpty(const moveit_msgs::msg::PlanningScene& msg);
-
-  /** \brief Check if a message includes any information about a planning scene world, or it is just a default, empty
-   * message. */
-  [[deprecated("Use moveit/utils/message_checks.h instead")]] static bool
-  isEmpty(const moveit_msgs::msg::PlanningSceneWorld& msg);
-
-  /** \brief Check if a message includes any information about a robot state, or it is just a default, empty message. */
-  [[deprecated("Use moveit/utils/message_checks.h instead")]] static bool
-  isEmpty(const moveit_msgs::msg::RobotState& msg);
```
Comment (AndyZe) [inline moveit_core/planning_scene/include/moveit/planning_scene/planning_scene.h:935]: It looks like you still need to delete the definition of these functions in `planning_scene/src/planning_scene.cpp`

File: moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h [hunk 3]
```diff
@@ -155,8 +155,6 @@ class RobotTrajectory
    */
   double getWayPointDurationFromStart(std::size_t index) const;
 
-  [[deprecated]] double getWaypointDurationFromStart(std::size_t index) const;
```
Comment (AndyZe) [inline moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h:158]: Still need to delete the definition of this function from `robot_trajectory.cpp`

**Comment by AndyZe** [comment]
I rebased this branch so please do a `git pull` before resuming work.  👍

**Comment by AdarshSingh-14** [comment]
@AndyZe I have made the changes there should be no error now.

**Comment by AndyZe** [comment]
There's a known issue with Rolling CI right now, so don't expect that CI job to pass. Hopefully the other ones will.

**Review by henningkayser — COMMENTED** [review]
CI is fixed now, everything should work once you update the branch. Before merging, could you take care of the now outdated documentation inside PlanningSceneMonitor and MoveItCpp?

* ~https://github.com/ros-planning/moveit2/pull/1935/files#diff-cae47857f92b275bba3b5eac239e84b6735ab2cd597ef5446bc7cad077d33931L165-L172 should be removed~
* ~https://github.com/ros-planning/moveit2/pull/1935/files#diff-25aa49a70772f7d00842cd7a2614d94b2afefeb84ceceb0cb5cdc2272c11fe5bL119-L123, https://github.com/ros-planning/moveit2/pull/1935/files#diff-25aa49a70772f7d00842cd7a2614d94b2afefeb84ceceb0cb5cdc2272c11fe5bL133-L137, https://github.com/ros-planning/moveit2/pull/1935/files#diff-25aa49a70772f7d00842cd7a2614d94b2afefeb84ceceb0cb5cdc2272c11fe5bL147-L152, https://github.com/ros-planning/moveit2/pull/1935/files#diff-25aa49a70772f7d00842cd7a2614d94b2afefeb84ceceb0cb5cdc2272c11fe5bL163-L168 need to be updated~

See review below

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h [hunk 4]
```diff
@@ -121,12 +121,6 @@ class MOVEIT_PLANNING_SCENE_MONITOR_EXPORT PlanningSceneMonitor
    *  @param tf_buffer A pointer to a tf2_ros::Buffer
    *  @param name A name identifying this planning scene monitor
    */
-  [[deprecated("Passing tf2_ros::Buffer to PlanningSceneMonitor's constructor is deprecated")]] PlanningSceneMonitor(
```
Comment (henningkayser) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h:124]: fix

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h [hunk 5]
```diff
@@ -135,12 +129,6 @@ class MOVEIT_PLANNING_SCENE_MONITOR_EXPORT PlanningSceneMonitor
...
```
Comment (henningkayser) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h:138]: fix

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h [hunk 6]
```diff
@@ -150,13 +138,6 @@ class MOVEIT_PLANNING_SCENE_MONITOR_EXPORT PlanningSceneMonitor
...
```
Comment (henningkayser) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h:153]: fix

File: moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h [hunk 7]
```diff
@@ -166,13 +147,6 @@ class MOVEIT_PLANNING_SCENE_MONITOR_EXPORT PlanningSceneMonitor
...
```
Comment (henningkayser) [inline moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h:169]: fix

File: moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h [hunk 8]
```diff
@@ -170,10 +159,6 @@ class MoveItCpp
    * a controller. The exact behavior of finding a controller depends on which MoveItControllerManager plugin is active.
    * \return moveit_controller_manager::ExecutionStatus::SUCCEEDED if successful
    */
-  [[deprecated(
```
Comment (henningkayser) [inline moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h:173]: remove

**Review by henningkayser — COMMENTED** [review]
Above links don't work... I tagged the needed doxygen changes inline

**Pull request closed without merging**